shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/LedgerState.hs

```diff
@@ -945,27 +816,6 @@ propWits (Just (Update (ProposedPPUpdates pup) _)) (GenDelegs _genDelegs) =
   where
     updateKeys = Map.keysSet pup ◁ _genDelegs
 
-validTx ::
-  ( Crypto crypto,
-    DSignable crypto (Hash crypto (TxBody crypto))
-  ) =>
-  Tx crypto ->
-  GenDelegs crypto ->
-  SlotNo ->
-  PParams ->
-  LedgerState crypto ->
-  Validity
-validTx tx d' slot pp l =
-  validRuleUTXO
-    ((_rewards . _dstate . _delegationState) l)
-    ((_stPools . _pstate . _delegationState) l)
-    ((_stkCreds . _dstate . _delegationState) l)
-    pp
-    slot
-    tx
-    (_utxoState l)
-    <> validRuleUTXOW tx d' l
-
```

Comment:
Awesome that we can pull this out of the library code!

Reply:
it's been bugging me for a while now!